### ISSUE TITLE
Got error when build on gitlab

### ISSUE THREAD

**Issue by rymesaint** [cirruslabs/docker-images-flutter#72]
i got this error when trying to build from gitlab CI but when on local its working properly no error

```
[0KRunning with gitlab-runner 13.1.0 (6214287e)
[0;m[0K  on Me Macbook 8dvSH-fo
[0;msection_start:1593590957:prepare_executor
[0K[0K[36;1mPreparing the "docker" executor[0;m
[0;m[0KUsing Docker executor with image cirrusci/flutter:stable ...
[0;m[0KPulling docker image cirrusci/flutter:stable ...
[0;m[0KUsing docker image sha256:c97aaeea37dce46d9803a372b619fe76158f4d18831812450dd7b00534485e69 for cirrusci/flutter:stable ...
[0;msection_end:1593590963:prepare_executor
[0Ksection_start:1593590963:prepare_script
[0K[0K[36;1mPreparing environment[0;m
[0;mRunning on runner-8dvsh-fo-project-3-concurrent-0 via MACs-MacBook-Pro.local...
section_end:1593590965:prepare_script
[0Ksection_start:1593590965:get_sources
[0K[0K[36;1mGetting source from Git repository[0;m
[0;m[32;1mFetching changes with git depth set to 50...[0;m
Reinitialized existing Git repository in /builds/edukasi/wss-mobile/.git/
[32;1mChecking out 12a2941e as master...[0;m

[32;1mSkipping Git submodules setup[0;m
section_end:1593590966:get_sources
[0Ksection_start:1593590966:step_script
[0K[0K[36;1mExecuting "step_script" stage of the job script[0;m
[0;m[32;1m$ flutter build aot -t lib/main_${TARGET_FILE}[0;m
Running "flutter pub get" in wss-mobile...                         67.0s
Building AOT snapshot in release mode (android-arm-release)...     

Compiler message:

lib/src/app.dart:6:8: Error: Error when reading 'lib/src/Notifier/SetupProvider.dart': No such file or directory

import 'package:wss_mobile/src/Notifier/SetupProvider.dart';

       ^

lib/src/Screens/Menus/sarprasMenu.dart:18:22: Error: The method 'Menu' isn't defined for the class 'SarprasMenu'.

 - 'SarprasMenu' is from 'package:wss_mobile/src/Screens/Menus/sarprasMenu.dart' ('lib/src/Screens/Menus/sarprasMenu.dart').

Try correcting the name to the name of an existing method, or defining a method named 'Menu'.


    _appServices.add(Menu(
                     ^^^^


lib/src/Screens/Menus/sarprasMenu.dart:35:22: Error: The method 'Menu' isn't defined for the class 'SarprasMenu'.
 - 'SarprasMenu' is from 'package:wss_mobile/src/Screens/Menus/sarprasMenu.dart' ('lib/src/Screens/Menus/sarprasMenu.dart').



Try correcting the name to the name of an existing method, or defining a method named 'Menu'.
    _appServices.add(Menu(
                     ^^^^

lib/src/Screens/Menus/studentMenu.dart:20:10: Error: 'Menu' isn't a type.

    List<Menu> _appServices = [];


         ^^^^
lib/src/Screens/Menus/studentMenu.dart:22:22: Error: The method 'Menu' isn't defined for the class 'StudentMenu'.


 - 'StudentMenu' is from 'package:wss_mobile/src/Screens/Menus/studentMenu.dart' ('lib/src/Screens/Menus/studentMenu.dart').
Try correcting the name to the name of an existing method, or defining a method named 'Menu'.

    _appServices.add(Menu(


                     ^^^^
lib/src/Screens/Menus/studentMenu.dart:32:22: Error: The method 'Menu' isn't defined for the class 'StudentMenu'.

 - 'StudentMenu' is from 'package:wss_mobile/src/Screens/Menus/studentMenu.dart' ('lib/src/Screens/Menus/studentMenu.dart').

lib/src/Screens/Attendance/TeacherAttendanceList.dart:171:22: Error: The method 'HexColor' isn't defined for the class '_TeacherAttendanceList'.

 - '_TeacherAttendanceList' is from 'package:wss_mobile/src/Screens/Attendance/TeacherAttendanceList.dart' ('lib/src/Screens/Attendance/TeacherAttendanceList.dart').

Try correcting the name to the name of an existing method, or defining a method named 'HexColor'.

    Color bgStatus = HexColor(teacher.status.hexColor);

                     ^^^^^^^^

lib/src/Screens/Attendance/TeacherAttendanceList.dart:276:30: Error: The method 'HexColor' isn't defined for the class '_TeacherAttendanceList'.

 - '_TeacherAttendanceList' is from 'package:wss_mobile/src/Screens/Attendance/TeacherAttendanceList.dart' ('lib/src/Screens/Attendance/TeacherAttendanceList.dart').

Try correcting the name to the name of an existing method, or defining a method named 'HexColor'.

            backgroundColor: HexColor(color),

                             ^^^^^^^^


lib/src/Screens/Attendance/kepsek_prence_history_student.dart:32:3: Error: 'Student' isn't a type.
  Student _selectedStudent;

  ^^^^^^^

lib/src/Screens/Attendance/kepsek_prence_history_student.dart:33:3: Error: 'ClassModel' isn't a type.


  ClassModel _selectedClass;
  ^^^^^^^^^^

lib/src/Screens/Attendance/kepsek_prence_history_student.dart:34:10: Error: 'ResultStudentPresence' isn't a type.


  Future<ResultStudentPresence> _presence;
         ^^^^^^^^^^^^^^^^^^^^^

lib/src/Screens/Attendance/kepsek_prence_history_student.dart:35:10: Error: 'StudentListModel' isn't a type.

  Future<StudentListModel> _listStudent;

         ^^^^^^^^^^^^^^^^

lib/src/Screens/Attendance/kepsek_prence_history_student.dart:36:10: Error: 'ClassListModel' isn't a type.

  Future<ClassListModel> _classes;

         ^^^^^^^^^^^^^^

lib/src/Screens/Attendance/kepsek_prence_history_student.dart:94:10: Error: 'ClassModel' isn't a type.

    List<ClassModel> classes = snapshots.data.listClass;

         ^^^^^^^^^^

lib/src/Screens/Attendance/kepsek_prence_history_student.dart:98:29: Error: 'ClassModel' isn't a type.

      child: DropdownButton<ClassModel>(

                            ^^^^^^^^^^

lib/src/Screens/Attendance/kepsek_prence_history_student.dart:102:29: Error: 'ClassModel' isn't a type.

        items: classes.map((ClassModel classe) {
                            ^^^^^^^^^^
lib/src/Screens/Attendance/kepsek_prence_history_student.dart:103:35: Error: 'ClassModel' isn't a type.
          return DropdownMenuItem<ClassModel>(
                                  ^^^^^^^^^^





lib/src/Screens/Attendance/kepsek_prence_history_student.dart:113:21: Error: 'ClassModel' isn't a type.

        onChanged: (ClassModel selected) {

                    ^^^^^^^^^^

lib/src/Screens/Attendance/kepsek_prence_history_student.dart:126:15: Error: 'ClassModel' isn't a type.

  getStudents(ClassModel selected) {

              ^^^^^^^^^^

lib/src/Screens/Attendance/kepsek_prence_history_student.dart:169:10: Error: 'Student' isn't a type.

    List<Student> students = snapshots.data.listStudent;

         ^^^^^^^

lib/src/Screens/Attendance/kepsek_prence_history_student.dart:173:29: Error: 'Student' isn't a type.

      child: DropdownButton<Student>(

                            ^^^^^^^

lib/src/Screens/Attendance/kepsek_prence_history_student.dart:177:30: Error: 'Student' isn't a type.

        items: students.map((Student student) {

                             ^^^^^^^

lib/src/Screens/Attendance/kepsek_prence_history_student.dart:178:35: Error: 'Student' isn't a type.

          return DropdownMenuItem<Student>(
                                  ^^^^^^^


lib/src/Screens/Attendance/kepsek_prence_history_student.dart:188:21: Error: 'Student' isn't a type.


        onChanged: (Student selected) {
                    ^^^^^^^


lib/src/Screens/Attendance/kepsek_prence_history_student.dart:248:38: Error: 'Student' isn't a type.
  loadMarker(DateTime selectedMonth, Student student) async {

                                     ^^^^^^^

lib/src/Screens/Permission/widgets/LeaveCardInfo.dart:30:26: Error: The method 'getColorStatus' isn't defined for the class 'LeaveCardInfo'.

 - 'LeaveCardInfo' is from 'package:wss_mobile/src/Screens/Permission/widgets/LeaveCardInfo.dart' ('lib/src/Screens/Permission/widgets/LeaveCardInfo.dart').

Try correcting the name to the name of an existing method, or defining a method named 'getColorStatus'.

                  color: getColorStatus(permission),

                         ^^^^^^^^^^^^^^

lib/src/Screens/Permission/widgets/VerifiedLeavePermission.dart:10:9: Error: 'LeavePermission' isn't a type.

  final LeavePermission permission;

        ^^^^^^^^^^^^^^^

lib/src/Screens/Permission/widgets/VerifiedLeavePermission.dart:16:37: Error: 'AccountTypeProvider' isn't a type.

  Widget getDurationByPicketTeacher(AccountTypeProvider account, LeavePermission permission) {

                                    ^^^^^^^^^^^^^^^^^^^


lib/src/Screens/Permission/widgets/VerifiedLeavePermission.dart:16:66: Error: 'LeavePermission' isn't a type.

  Widget getDurationByPicketTeacher(AccountTypeProvider account, LeavePermission permission) {
                                                                 ^^^^^^^^^^^^^^^


lib/src/Screens/Permission/widgets/VerifiedLeavePermission.dart:34:33: Error: 'AccountTypeProvider' isn't a type.
    final account = Provider.of<AccountTypeProvider>(providerContext, listen: false);

                                ^^^^^^^^^^^^^^^^^^^

lib/src/Screens/Permission/widgets/VerifiedLeavePermission.dart:35:33: Error: 'VerifyLeavePermissionProvider' isn't a type.

    final provide = Provider.of<VerifyLeavePermissionProvider>(providerContext, listen: false);

                                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^






lib/src/Screens/Elearning/PreTest.dart:22:9: Error: 'PreTestModel' isn't a type.
  final PreTestModel pretest;
        ^^^^^^^^^^^^
lib/src/Screens/Elearning/PreTest.dart:35:3: Error: 'QuestionList' isn't a type.
  QuestionList _questionList;
  ^^^^^^^^^^^^

lib/src/Screens/Elearning/PreTest.dart:86:5: Error: 'AnswerList' isn't a type.

    AnswerList answerList = await _elearningController.getAllAnswerTemp(widget.pretest.id, refresh: refresh);

    ^^^^^^^^^^

lib/src/Screens/Elearning/PreTest.dart:87:36: Error: 'Question' isn't a type.
    _questionList.results.forEach((Question question) {


                                   ^^^^^^^^

lib/src/Screens/Elearning/PreTest.dart:88:33: Error: 'Answer' isn't a type.

      question.answers.forEach((Answer answer) async {

                                ^^^^^^

lib/src/Screens/Elearning/PreTest.dart:90:39: Error: 'Answer' isn't a type.

          answerList.answers.forEach((Answer answerTemp) {
                                      ^^^^^^
lib/src/Screens/Elearning/PreTest.dart:187:25: Error: 'Question' isn't a type.
  Widget questionWidget(Question question) {
                        ^^^^^^^^
lib/src/Screens/Elearning/PreTest.dart:234:23: Error: 'Answer' isn't a type.
  Widget optionAnswer(Answer answer, Question question, int questionNumber) {
                      ^^^^^^








lib/src/Screens/Elearning/PreTest.dart:234:38: Error: 'Question' isn't a type.

  Widget optionAnswer(Answer answer, Question question, int questionNumber) {

                                     ^^^^^^^^

lib/src/Screens/Elearning/PreTest.dart:363:44: Error: 'Question' isn't a type.

          tabs: _questionList.results.map((Question question) {

                                           ^^^^^^^^

lib/src/Screens/Elearning/PreTest.dart:380:15: Error: 'Question' isn't a type.

              Question question = await _elearningController.getAnswerTemp(widget.pretest.id, _questionList.results[index].id);

              ^^^^^^^^

lib/src/Screens/Elearning/PreTest.dart:403:48: Error: 'Question' isn't a type.
          children: _questionList.results.map((Question question) {


                                               ^^^^^^^^


lib/src/Screens/Elearning/PreTest.dart:416:53: Error: 'Answer' isn't a type.
                    children: question.answers.map((Answer answer) {

                                                    ^^^^^^
Building AOT snapshot in release mode (android-arm-release)...        28.0s
Exception: Errors during snapshot creation: null
The aot build failed.
section_end:1593591065:step_script
[0K[31;1mERROR: Job failed: exit code 1
[0;m
```

and here my .gitlab-ci.yml

```
image: cirrusci/flutter:stable

stages:
    - test # All jobs related for testing such as Unit Test
    - build # All jobs related for building app for iOS and Android

test:android-bundle:
    stage: test
    script:
        - flutter build aot -t lib/main_${TARGET_FILE}
        - flutter analyze -t lib/main_${TARGET_FILE}

build:android-bundle: #Job name
    stage: build # kind of job
    before_script:
        - flutter doctor -v
        - flutter pub get
        - flutter clean
        - flutter pub cache repair
        - echo $KEYSTORE | base64 --decode > ./android/wss_new.jks
        - echo "storeFile=wss_new.jks" > ./android/key.properties
        - echo "storeFileLinux=wss_new.jks" > ./android/key.properties
        - echo "keyAlias=${KEY_ALIAS}" >> ./android/key.properties
        - echo "keyPassword=${KEYPASS}" >> ./android/key.properties
        - echo "storePassword=${KEYPASS}" >> ./android/key.properties
    script:
        - flutter build appbundle --flavor ${FLAVOR} -t lib/main_${TARGET_FILE}
    artifacts:
        paths:
            - ./build/app/outputs/bundle/${FLAVOR}Release/app.aab
        expire_in: 1 week
    only:
        - master
        - release
    when: manual
```

did i do wrong with my config?

**Comment by fkorotkov** [comment]
I think GitLab doesn't respect `USER` directive from the image and you'll need to `sudo` your commands. Seems like related to #45

**Comment by rymesaint** [comment]
@fkorotkov using sudo not working only get another error like operation not permitted and trying to solve like #45 its not working, do i need to wait for a new release of sudo?

**Comment by burtsevyg** [comment]
> @fkorotkov using sudo not working only get another error like operation not permitted and trying to solve like #45 its not working, do i need to wait for a new release of sudo?

Can you show some test project on gitlab.com?

**Comment by fkorotkov** [comment]
Should be fixed as part of #93
